### PR TITLE
make sure that transformer only touches file that need to be rewritten

### LINKED ISSUE #26: transformer also transforms html files in lib
The transformer also transforms html files in lib, this can give problems together with angular2 for example. It would be better I think if this would only transform files in web, and also only change the html if there is a match.

it now adds html/body tags to angular templates for example:

### PR DESCRIPTION
**Pull request by kasperpeulen**
fixes #26 

kind of rewrote a bit more in the end, just for fun, added also some unit tests 

I can make this PR a little bit less drastic if you guys prefer that

### CODE REVIEW

**Comment by trentgrover-wf** [comment]
thanks for the PR @kasperpeulen , didn't realize it was here...

please review:
@maxwellpeterson-wf 
@evanweible-wf 
@jayudey-wf 
@greglittlefield-wf

File: pubspec.yaml [hunk 1]
```diff
@@ -5,11 +5,17 @@ description: Replaces Dart script tags with JavaScript script tags,
...
-  barback: ">=0.15.0 <0.16.0"
-  html: ">=0.12.0 <0.13.0"
+  barback: '^0.15.2+7'
+  html: '^0.12.2+1'
 dev_dependencies:
-  browser: any
-  test: any
+  browser: '^0.10.0+2'
+  test: '^0.12.6+2'
+  grinder: '^0.8.0+1'
```
Comment (evanweible-wf) [inline pubspec.yaml]: Doesn't look like grinder is actually being used, does this need to be here?

**Comment by evanweible-wf** [comment]
@kasperpeulen thanks for the PR! Looks good, I just have the one question

File: pubspec.yaml (hunk 1 above)
Comment (kasperpeulen) [inline pubspec.yaml]: I use grinder for an easy way to run pub build in the e2e test:
https://github.com/kasperpeulen/dart_to_js_script_rewriter/blob/98956b6d7ee41875458c1248cea5c69bf7e03fbc/test/end2end_test.dart#L2

**Comment by rmconsole2-wf** [comment]
@kasperpeulen This pull request has merge conflicts, please resolve.

**Comment by kasperpeulen** [comment]
@rmconsole2-wf This PR is (or was) a complete rewrite of the current code. It would make much more sense to first let this PR in, and then merge other PR's if you ask me.

**Comment by maxwellpeterson-wf** [comment]
@kasperpeulen it cannot be merged now because there are merge conflicts with master. You'll need to rebase or merge master into this branch and fix the conflicts.

**Comment by trentgrover-wf** [comment]
@maxwellpeterson-wf since we stomped on his PR, let's go ahead and fix the conflicts ourselves with a PR against his branch

**Comment by evanweible-wf** [comment]
Sorry, this still shows merge conflicts. I've opened up a new PR with this branch rebased against the latest master: #29

**Comment by trentgrover-wf** [comment]
closing this and will review under #29 (still retains all of @kasperpeulen 's commits)

**Pull request closed without merging**